### PR TITLE
chore: drop `scale_name` ggplot2 arg and drop fansi dep

### PR DESCRIPTION
**Pull request by olivroy**
Drop fansi dep and avoid setting scale_name in ggplot2

### CODE REVIEW

**Comment by olivroy** [comment]
@krlmlr since cli provides a lot of the ansi functions, would you consider dropping fansi? or is the performance loss high?

File: R/extent.R [hunk 1]
```diff
@@ -5,15 +5,15 @@
...
+#' @importFrom cli utf8_nchar
 #' @examples
 #' get_extent(c("abc", "de"))
 #' get_extent("\u904b\u6c23")
 get_extent <- function(x) {
   force(x)
-  x <- strip_sgr(x, warn = FALSE)
-  width <- utf8_width(x, encode = FALSE, utf8 = TRUE)
+  x <- ansi_strip(x)
+  width <- cli::utf8_nchar(x, "width")
```
Comment (krlmlr) [inline R/extent.R:16]:
```suggestion
  width <- utf8_nchar(x, "width")
```

**Review by krlmlr — APPROVED** [review]
Thanks for taking this on! No objections, but tibble uses fansi too IIUC.

**Review by krlmlr — APPROVED** [review]
Thanks for taking this on! No objections, but tibble uses fansi too IIUC.

**Comment by olivroy** [comment]
I will open a PR in tibble and test them both!

**Pull request merged**